source/kraken/apply_disruption.cpp

```diff
@@ -125,14 +147,43 @@ struct add_impacts_visitor : public apply_impacts_visitor {
 
     using apply_impacts_visitor::operator();
 
-    void operator()(nt::MetaVehicleJourney* mvj, const nt::Route* r = nullptr) {
+    void operator()(nt::MetaVehicleJourney* mvj, nt::Route* r = nullptr) {
         log_start_action(mvj->uri);
         if (impact->severity->effect == nt::disruption::Effect::NO_SERVICE) {
             LOG4CPLUS_TRACE(log, "canceling " << mvj->uri);
             mvj->cancel_vj(rt_level, impact->application_periods, pt_data, meta, r);
             mvj->impacted_by.push_back(impact);
+        } else if (impact->severity->effect == nt::disruption::Effect::MODIFIED_SERVICE) {
+            LOG4CPLUS_TRACE(log, "modifying " << mvj->uri);
+            auto vp = compute_vp(impact->application_periods, meta.production_date);
+            if (! r && ! mvj->get_base_vj().empty()) {
+                r = mvj->get_base_vj().at(0)->route;
+                LOG4CPLUS_TRACE(log, mvj->get_base_vj().at(0)->validity_patterns[0]->str());
```

Comment:
I think this trace is only for debug no ? else there should be a bit more explanation